src/main/scala/org/ergoplatform/local/ErgoMiner.scala

```diff
@@ -322,20 +321,35 @@ class ErgoMiner(ergoSettings: ErgoSettings,
       tx -> cost
     }
 
+    val poolTxs = pool.getAllPrioritized
+
     val (txs, toEliminate) = ErgoMiner.collectTxs(minerPk,
       state.stateContext.currentParameters.maxBlockCost,
       state.stateContext.currentParameters.maxBlockSize,
       maxTransactionComplexity,
       state,
       upcomingContext,
-      pool.getAllPrioritized,
+      poolTxs,
       emissionTxOpt.toSeq)(state.stateContext.validationSettings)
 
     // remove transactions which turned out to be invalid
     if (toEliminate.nonEmpty) viewHolderRef ! EliminateTransactions(toEliminate)
 
-    state.proofsForTransactions(txs).map { case (adProof, adDigest) =>
-      CandidateBlock(bestHeaderOpt, version, nBits, adDigest, adProof, txs, timestamp, extensionCandidate, votes)
+    state.proofsForTransactions(txs) match {
+      case Success((adProof, adDigest)) =>
+        Success(CandidateBlock(bestHeaderOpt, version, nBits, adDigest, adProof, txs, timestamp, extensionCandidate, votes))
+      case Failure(t: Throwable) =>
+        emissionTxOpt match {
+          case Some(emissionTx) =>
+            log.error("Failed to produce proofs for transactions, but emission box is found: ", t)
+            val fallbackTxs = Seq(emissionTx._1)
```

Comment:
Please add description, why emissionTx is used as a fallback.

Reply:
Done